### ISSUE TITLE
Restore gangamon server

### ISSUE THREAD

**Issue by rob-c** [ganga-devs/ganga#643]
OK, I need to get around to restoring the gangamon server in order to restore the `report` functionality.
I'm going to list the stages which need looking into here
- [x] Update the code on github with the modified code from the current collection server to Centos7 compatible. (systemd service+new Django+new crontask+new Apache2 config etc.)
- [x] Launch a new machine monitored and updates via Jenkins and Foreman
- [ ] Write some new Jenkins rules to pull the code from github, check the secrets from storage then update the machine.
- [ ] Edit: Update security to look for secrets stored on gangage account via aiadm. (May require remote login with gangage)
- [ ] Make sure the machine can be recreated from scratch using this recipe.
- [ ] Writeup how it currently works and how the pieces fit together.
- [ ] Edit: Setup the new hostcert to be allowed to pull from CERN messaging system.
- [ ] Change the DNS target which could prove difficult if the rule points to no machine?
- [ ] Get the machine released from behind the CERN firewall (ports 443 and 80 are all that's required I think).
- [ ] Edit: Re-write the collector to not have to be restarted once every 30min.

After this has been done I'll look into fixing/expanding the `report` functionality and we can consider adding some more information to the stomp reporting.
(The current implementation of the listener may require scrapping eventually and being replaced with a more reliable python script with direct SQL commands. Although the DJango saves from a lot of this it doesn't scale very well in it's current form at all.)

**Comment by rob-c** [comment]
I've had to fix some minor issues on the Jenkins descriptions.
Does anyone know if we're supposed to get some Foreman alarm when Jenkins fails to apply to these boxes?
It would be better than waiting for someone higher up to notice the boxes haven't been updating and are potentially out of date.
(I'll have a look at the configuration and I will check all of the machines are managed by the CERN group which should give the appropriate users messages about this).

OK, so far I've managed to upload the changes required for CC7 compatibility to github. I'm in the process of sorting out the Jenkins rules into what is and isn't CC7 friendly and then I'll update the secrets to be stored in a new location vs the secrets being stored on a single instance or elsewhere.
(I think the CI and Worker scripts can probably inherit from some additional common script but I'm more keen to get the Jenkins recipies working first then I'll worry about cleaning them afterwards.)

At this point I'm tempted to add and save a common shared keypair for managing all machines on the CERN openstack should only the gangage account be able to access them in future. (As well as documenting what secrets are where).

I may migrate the ganga databases to be managed by the ganga Jenkins group for simplicity but this is a final step after I fix this server instance.

I still plan to re-implement the collector if I get a chance as the current implementation only works because it gets killed once every 30min due to a memory leak due to poor design. (Again a final step on the list). The fact this service has to be restarted so often is why I'm strongly moving for this server to be CC7 based so that I have systemd which is better than sysvinit on this regard.
